cmp/store/impl/StoreFilterFieldImplModel.js

```diff
@@ -117,11 +118,16 @@ export class StoreFilterFieldImplModel {
     // Implementation
     //------------------------
     applyFilter() {
-        this.store?.filterModel.replaceFilters(this.filter);
+        if (this.store) {
+            const {filterModel} = this.store;
+            const newFilters = filter(filterModel.filters, {group: this.xhId});
+            if (this.filter) newFilters.push(this.filter);
+            filterModel.setFilters(newFilters);
```

Comment:
I'm not sure I understand this code... are we ever expecting to have more than 1 filter in this group? Also, by using setFilters here, we're removing any filters *not* in this group.

Wouldn't it be clearer to `removeFiltersByGroup` and then `addFilter(this.filter)`? This will preserve any filters not in this group and ensure that the StoreFilterField only accounts for one filter in the FilterModel

Reply:
you are understanding this correctly tom!  It was bust -- just committed a fix.

Re:  not doing two separate operations -- Was trying to be transactional about it, although I suppose wrapping it in an action might have been effectively the same thing.